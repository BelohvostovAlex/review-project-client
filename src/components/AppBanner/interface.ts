export interface AppBannerProps {
  title: string;
  text?: string;
}
