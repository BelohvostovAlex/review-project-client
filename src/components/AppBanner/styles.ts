export const makeStyles = () => ({
  bannerWrapper: {
    width: "600px",
    height: "400px",
    border: "1px solid black",
    borderRadius: "10px",
    boxSizing: "border-box",
    padding: "60px",
    margin: "50px auto",
  },
  bannerTitle: {
    textAlign: "center",
    marginBottom: "30px",
  },
});
