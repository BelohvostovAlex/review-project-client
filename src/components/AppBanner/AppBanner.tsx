import { FunctionComponent } from "react";

import { Box, Typography } from "@mui/material";

import { AppBannerProps } from "./interface";
import { makeStyles } from "./styles";

export const AppBanner: FunctionComponent<AppBannerProps> = ({
  title,
  text,
}) => {
  const style = makeStyles();
  return (
    <Box sx={style.bannerWrapper}>
      <Typography sx={style.bannerTitle}>{title}</Typography>
      <Box component="p">{text}</Box>
    </Box>
  );
};
