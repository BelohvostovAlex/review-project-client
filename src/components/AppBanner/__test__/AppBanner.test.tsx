import { render, screen } from "@testing-library/react";
import { AppBanner } from "../AppBanner";

import "@testing-library/jest-dom";

describe("backdrop", () => {
  it("AppBanner exists with only title prop", async () => {
    render(<AppBanner title="AppBanner title" />);
    const appBannerTitle = screen.getByText(/appbanner title/i);

    expect(appBannerTitle).toBeInTheDocument();
    expect(appBannerTitle).toMatchSnapshot();
  });

  it("AppBanner exists with title,text props", async () => {
    render(<AppBanner title="AppBanner title" text="AppBanner text" />);
    const appBannerTitle = screen.getByText(/appbanner title/i);
    const appBannerText = screen.getByText(/appbanner text/i);

    expect(appBannerTitle).toBeInTheDocument();
    expect(appBannerText).toBeInTheDocument();
  });
});
