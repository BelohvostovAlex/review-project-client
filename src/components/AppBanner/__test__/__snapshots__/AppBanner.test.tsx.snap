// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backdrop AppBanner exists with only title prop 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-1b4m6mh-MuiTypography-root"
>
  AppBanner title
</p>
`;
